src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1369,12 +1369,42 @@ private BoundExpression BindNonMethod(SimpleNameSyntax node, Symbol symbol, Diag
                         }
                         else
                         {
+                            SourceLocalSymbol sourceLocal;
+                            type = null;
+
+                            if (node.SyntaxTree == localSymbolLocation.SourceTree && (object)(sourceLocal = localSymbol as SourceLocalSymbol) != null &&
```

Comment:
What's the difference with the simpler check "sourceLocal is SourceLocalSymbol"?